static/app/components/core/segmentedControl/index.tsx

```diff
@@ -143,21 +151,52 @@ function Segment<Value extends string>({
   const showDivider = !isSelected && !nextOptionIsSelected;
 
   const {isDisabled} = props;
+
+  const label = theme.isChonk ? (
+    <VisibleLabel
+      isSelected={isSelected}
+      isDisabled={isDisabled}
+      priority={priority}
+      role="presentation"
+    >
+      {props.children}
+    </VisibleLabel>
+  ) : (
+    // Once an item is selected, it gets a heavier font weight and becomes slightly
+    // wider. To prevent layout shifts, we need a hidden container (HiddenLabel) that
+    // will always have normal weight to take up constant space; and a visible,
+    // absolutely positioned container (VisibleLabel) that doesn't affect the layout.
```

Comment:
none of this madness is needed in chonk because selected elements don’t get wider so we can get rid of those wrappers and the absolute positioning 🎉